### PR TITLE
feat: devex improvements

### PR DESCRIPTION
**Pull request by rpalakkal**
* adds compute proof/query checking on `run`
* adds ability to /sdk to override max user outputs and max user subqueries
* subquery caller caches rpc calls
* allows specifying custom circuit configuration (via cli)
* adds `--aggregate` to the cli
* replaces `gen_srs` with `read_srs_from_dir` (requires the correct params to be installed to `./params` or to a user specified path)

### CODE REVIEW

File: circuit/src/input/raw_input.rs [hunk 1]
```diff
@@ -62,12 +62,34 @@ impl<F: Field, const N: usize> RawInput<F> for [u8; N] {
...
+        res
+    }
+}
+
+impl<F: Field, const N: usize> RawInput<F> for [usize; N] {
+    type FEType<T: Copy> = [T; N];
+    fn convert(&self) -> Self::FEType<F> {
+        let mut res = [F::ZERO; N];
+        for i in 0..N {
+            res[i] = F::from_u128(self[i] as u128);
```
Comment (jonathanpwang) [inline circuit/src/input/raw_input.rs:81]: `F::from`

**Comment by rpalakkal** [comment]
this branch is merged into #13, so closing this

File: circuit/src/input/flatten.rs [hunk 2]
```diff
@@ -11,26 +11,30 @@ pub trait InputFlatten<T: Copy>: Sized {
...
+pub struct FixLenVec<T: Copy + Default, const N: usize>(pub Vec<T>);
+
+impl<T: Copy + Default, const N: usize> Default for FixLenVec<T, N> {
+    fn default() -> Self {
+        Self(vec![T::default(); N])
+    }
 }
 
-impl<T: Copy, const N: usize> FixLenVec<T, N> {
+impl<T: Copy + Default, const N: usize> FixLenVec<T, N> {
```
Comment (jonathanpwang) [inline circuit/src/input/flatten.rs:22]: I think the Default here is not necessary?

File: circuit/src/run/inner.rs [hunk 3]
```diff
@@ -21,78 +18,73 @@ use crate::{
 };
 
 pub fn mock<P: JsonRpcClient + Clone, S: AxiomCircuitScaffold<P, Fr>>(
```
Comment (jonathanpwang) [inline circuit/src/run/inner.rs:20]: could this be a function of `AxiomCircuit` now? just a nice-to-have

**Review by jonathanpwang — APPROVED** [review]
LGTM, but make a changelog of all user-facing interface changes, so we can add to future release notes and also to docs.

**Pull request merged**